### ISSUE TITLE
Events on ETH side give error and so does trace-packets

### ISSUE THREAD

**Issue by tmsdkeys** [open-ibc/ibc-sdk#172]
The events on eth-execution don't work (after having successfully having created channel and sent packet(s) over):

```bash
ibctl events eth-execution
>
11 : OwnershipTransferred
70 : OwnershipTransferred
80 : OpenIbcChannel
[11:57:07.939 error]: no matching event (argument="topichash", value="0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef", code=INVALID_ARGUMENT, version=abi/5.7.0)
```

and same thing when attempting to trace packets:

```bash
ibctl trace-packets $'eth-execution:channel-0:polyibc.Ethereum-Devnet.B10c73e50B9bdB51f3504F7104a411174B9C3aa3' $'wasm:channel-0:wasm.wasm.wasm14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s0phg4d'
>
[11:59:51.381 error]: Error: no matching event (argument="topichash", value="0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef", code=INVALID_ARGUMENT, version=abi/5.7.0)
```

**Comment by dshiell** [comment]
Happens only with 2nd quickstart tutorial

**Comment by tmsdkeys** [comment]
Update: seems like the issue is on ethers.js side, has to do with events parsing in custom contracts (when I disable events being emitted the issue is resolved)

so no direct SDK issue but will likely still impact xdapp devs using hardhat/ethers

reference: https://github.com/ethers-io/ethers.js/issues/733

**Comment by nicopernas** [comment]
I can confirm that ethers-js v6 fixes this problem (we are currently using v5.7.2) However, the migration is not trivial. I have a wip branch (`npernas/migration-to-ethers6`) that confirmed the fix but I'd like to spend more time on that before merging. 

I can apply a similar fix to the sdk (based on the linked ethers issue) which also solves the problem. Something along the lines of 
```diff
diff --git a/src/lib/query.ts b/src/lib/query.ts
index 05fa21b..cb88f11 100644
--- a/src/lib/query.ts
+++ b/src/lib/query.ts
@@ -358,7 +358,12 @@ async function evmEvents(chain: EvmChainSet, opts: EventsFilter, cb: TxEventCb)
 
   let event: TxEvent = { height: logs[0].blockNumber, events: {} }
   for (const l of logs) {
-    const parsed = iface.parseLog(l)
+    let parsed: ethers.utils.LogDescription
+    try {
+      parsed = iface.parseLog(l)
+    } catch {
+      continue
+    }
     if (l.blockNumber === event.height) {
       event.events[parsed.name] = parse(parsed)
       continue
```

I am inclined to go after this approach

**Comment by nicopernas** [comment]
https://github.com/open-ibc/ibc-sdk/pull/187
